webapp/components/DataQuery/Visualizer/Chart/Chart.js

```diff
@@ -0,0 +1,189 @@
+
+ Maybe the only thing that the user needs to chose on the left pannel is the last entity
+*/
+
+const defaultSpec = {
+  $schema: 'https://vega.github.io/schema/vega-lite/v5.json',
+  /*name: 'tree_health_label',
+  description: 'tree_health_label',
+  title: {
+    text: 'tree_health_label',
+  },*/
+  /*width: 500,
+  height: 500,
+  autosize: {
+    type: 'fit',
+    contatins: 'padding',
+    resize: true,
+  },*/
+  /*data: {
+    name: 'table',
+    values: [],
+  },*/
+
+  layer: [
+    {
+      mark: { type: 'arc', innerRadius: 60, outerRadius: 80 },
+    },
+    /*{
+    mark: {type: "text", radius: 120},
+    encoding: {
+      text: {field: "tree_health_label", type: "nominal"}
+    }
+  }*/
+  ],
+  encoding: {
+    theta: {
+      field: 'tree_health_label',
+      type: 'nominal',
+      aggregate: 'count',
+      impute: {
+        value: 'NULL',
+      },
+      //stack: true
+    },
+    color: {
+      field: 'tree_health_label',
+      type: 'nominal',
+      /*legend: {
+        title: 'Tree health',
+      },*/
+
+      impute: {
+        value: 'NULL',
+      },
+    },
+    /*column: {
+      field: 'tree_origin_label',
+      type: 'nominal',
+      impute: {
+        value: 'NULL',
+      },
+    },
+    row: {
+      field: 'cluster_accessibility_label',
+      type: 'nominal',
+      impute: {
+        value: 'NULL',
+      },
+    },*/
+  },
+}
+
+const updateCategoyOnDonnutChart =
+  ({ chart: _chart }) =>
+  ({ nodeDef }) => {
+    return {
+      ..._chart,
+      encoding: {
+        ..._chart.encoding,
+        theta: { ..._chart.encoding.theta, field: nodeDef.name },
+        color: { ..._chart.encoding.color, field: nodeDef.name },
+      },
+    }
+  }
+
+const Chart = (props) => {
+  const i18n = useI18n()
+
+  const [chart, setChart] = useState(defaultSpec)
+  const [chartImage, setChartImage] = useState(null)
+
+  const [showChartEditor, setShowChartEditor] = useState(true)
+
+  const surveyId = useSurveyId()
+  const cycle = useSurveyCycleKey()
+
+  const { query } = props
+  useEffect(() => {
+    const getChart = async () => {
+      try {
+        const { data } = await axios.post(getUrl({ surveyId, query }), {
+          cycle,
+          query: A.stringify(query),
+          chart: A.stringify(chart),
+        })
+        setChartImage(data.svg)
+      } catch (err) {
+        console.log(err)
+      }
+    }
+    getChart()
+    return () => {}
+  }, [surveyId, cycle, query, chart])
+
+  return (
+    <div className="chart_container">
+      <button onClick={() => setShowChartEditor(true)}>Show pannel</button>
+      {chartImage && (
+        <img
+          src={`data:image/svg+xml;base64,${btoa(unescape(encodeURIComponent(chartImage)))}`}
+          alt=""
+          width="100%"
+          height="100%"
+        />
+      )}
+      {showChartEditor && (
+        <PanelRight width="30vw" onClose={() => setShowChartEditor(false)} header={i18n.t('appModules.categories')}>
```

Comment:
this is going to be a panel with 3 modes:

basic -> chose between templates and allow the user to modify some basic fields:
f.e: in arc chart choose the radius, the nodeDef to categorize and if there are some groups the nodeDef of the group
custom -> [ customize the charts based on different properties -> the most complex to implement ]
the most complex to implement ]
advanced -> define the spec using a JSON editor


Now we have a panel on the left to choose the entity and the columns, but now the only thing needed is the entity so maybe we should replace that.

Also maybe is a good idea to try to do this panelRight agnostic of "nodeDefs" I mean give some way to get the variables from outside.